shell/platform/android/flutter_main.cc

```diff
@@ -75,6 +92,23 @@ void FlutterMain::Init(JNIEnv* env,
 
   auto settings = SettingsFromCommandLine(command_line);
 
+  // Turn systracing on if ATrace_isEnabled is true AND
```

Comment:
I am worried that as we add to these cases here, reasoning about when tracing to systrace will actually be enabled is going to be get hard to reason about. For instance, how is the user to know that sytraceing is incompatible with startup tracing? Seems like a reasonable thing to attempt.

Reply:
Hmm. I can remove the trace_startup logic for sure. I thought we did something weird with the VM arguments in this case, but it's actually just the tool that causes the app to exit after startup, not the engine.

The main problem is that in release mode, none of these flags are passed typically - the Flutter tooling does not set any intents in release mode for this, and even if it did many applications strip out such debugging related intents in release mode.

Are you ok with this if it's just simplified down to a warning if you ask for the endless recorder and this happens, mentioning that the size of the buffer will be controlled by the systrace invocation rather than by us?

Reply:
I've simplified this significantly. I'm still checking whether ATrace is enabled because I don't like the idea to defaulting it to on for release - it seems like it may impact performance unnecessarily.